### PR TITLE
feat(kubeconform): allow using custom resource schemas

### PR DESCRIPTION
**Pull request by martin-sucha**
Allowing to pass custom -schema-location arguments enables us to
validate custom Kubernetes resources.

### CODE REVIEW

**Comment by mrVanboy** [comment]
Could you plz elaborate in commit msg why the custom wrapper is required please? 🙏

File: kubeconform/analyse.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+def analyse_overlay(overlay_dir):
+    print(f"🧐 {overlay_dir}: \t", end='', flush=True)
+
+    extra_args = []
+    if args.schema_location:
+        for location in args.schema_location:
+            extra_args.extend(['-schema-location', location])
+
+    p1 = subprocess.Popen(["kubectl", "kustomize", overlay_dir], stdout=subprocess.PIPE)
+    p2 = subprocess.Popen(["kubeconform", "--summary", "--exit-on-error", "--cache", kubeconform_cache] + extra_args,
```
Comment (aexvir) [inline kubeconform/analyse.py:33]: according [the docs](https://github.com/yannh/kubeconform) all the flags should have a single dash, but I guess it accepts both seeing that before they were mixed...

I'd rather use the same style everywhere, so
```suggestion
    p2 = subprocess.Popen(["kubeconform", "-summary", "-exit-on-error", "-cache", kubeconform_cache] + extra_args,
```

File: kubeconform/analyse.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import tempfile
+import argparse
+import os.path
+import sys
+from pathlib import Path
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('analyse_dir', nargs='?', default='.')
+parser.add_argument('--schema-location', action='append')
```
Comment (aexvir) [inline kubeconform/analyse.py:15]: and in that case why not also passing it the same way to our own cli?

File: kubeconform/analyse.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    if return_code != 0:
+        raise SystemExit(return_code)
+
+
+def analyse_k8s(dir):
+    overlays_dir = Path(dir) / 'overlays'
+    if overlays_dir.is_dir():
+        overlay_dirs = list(overlays_dir.glob('*'))
+    else:
+        overlay_dirs = list(Path(dir).glob('*/overlays/*'))
```
Comment (aexvir) [inline kubeconform/analyse.py:46]: 🤔 shouldn't an error be raised if the overlays dir is not a dir?
not sure about this default behaviour, at least without an error message

although I see that's what the previous bash script was doing... 😅

**Review by aexvir — APPROVED** [review]
nothing major preventing this to be merged, just a couple comments

File: kubeconform/analyse.py (hunk 2 above)
Comment (martin-sucha) [reply kubeconform/analyse.py:15]: AFAIK `argparse` only allows `--` for long options.

File: kubeconform/analyse.py (hunk 3 above)
Comment (martin-sucha) [reply kubeconform/analyse.py:46]: We want to support both `k8s/overlays/*` and `k8s/*/overlays/*` paths. The former used in simple projects, the latter in more complex ones.

**Comment by martin-sucha** [comment]
Squashed and updated the commit message.

File: kubeconform/analyse.py (hunk 2 above)
Comment (aexvir) [reply kubeconform/analyse.py:15]: should work just fine 🤔

**Comment by mrVanboy** [comment]
LGTM

File: kubeconform/analyse.py (hunk 2 above)
Comment (martin-sucha) [reply kubeconform/analyse.py]: Indeed it works. I learn something new every day :)

**Review by aexvir — APPROVED** [review]

**Pull request merged**